server/src/test/java/org/elasticsearch/cluster/metadata/AutoExpandReplicasTests.java

```diff
@@ -282,4 +282,16 @@ public void testOnlyAutoExpandAllocationFilteringAfterAllNodesUpgraded() {
             terminate(threadPool);
         }
     }
+
+    public void testCalculateDesiredNumberOfReplicas() {
+        String settingValue = randomFrom("0-all", "0-3");
```

Comment:
Can we explore the space a bit more here, for instance by choosing both lower and upper bound randomly like:

```
int lowerBound = between(0,9);
int upperBound = between(lowerBound, 10);
String settingValue = lowerBound + "-" + upperBound;
```

Having a dedicated test (or random variation) with "all" also makes sense.

Reply:
Done in https://github.com/elastic/elasticsearch/pull/87505/commits/ec7d7f97bc039f46d8c508943783b44ea78edae2.